prow/plugins/cherrypickunapproved/cherrypick-unapproved.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cherrypickunapproved adds the `do-not-merge/cherry-pick-not-approved`
+// label to PRs against a release branch which do not have the
+// `cherrypick-approved` label.
+package cherrypickunapproved
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "cherrypick-unapproved"
+
+const (
+	cpUnapprovedLabel         = "do-not-merge/cherry-pick-not-approved"
+	cpApprovedLabel           = "cherrypick-approved"
+	labelUnapprovedFormat     = "This PR is not for the master branch but does not have the `%s` label. Adding the `%s` label.\n"
+	assignPatchReleaseManager = `
+	To approve the cherrypick, please assign the patch release manager for the ` + "`%s`" + ` branch by writing ` + "`/assign @username`" + ` in a comment when ready.
+The list of patch release managers for each release can be found [here](https://git.k8s.io/sig-release/release-managers.md).`
```

Comment:
Oops, missed this -- should be configurable as well for other deployments. Can be done in a follow-up (or by me).

Reply:
> Oops, missed this -- should be configurable as well for other deployments. Can be done in a follow-up (or by me).

Updated in this PR. :)